treeherder/etl/schema.py

```diff
@@ -0,0 +1,28 @@
+import os
+import yaml
+from memoize import mproperty
+
+
```

Comment:
An alternative implementation coul be to just use module attributes:
```python
def get_json_schema(schema_name):
    file_path = os.path.join("schemas", filename, ".yml")
    with open(file_path) as f:
        return yaml.load(f)

text_log_summary_artifact = get_json_schema("text-log-summary-artifact")
```
and then it could be imported with:
```python
from treeherder.etl.schema import text_log_summary_artifact
```
I find this solution easier to understand and to use, but I'll leave the final decision to you.